frontend/src/main/scala/bloop/dap/BloopDebuggeeRunner.scala

```diff
@@ -138,10 +146,157 @@ object BloopDebuggeeRunner {
   ): Either[String, DebuggeeRunner] = {
     projects match {
       case Seq() => Left(s"No projects specified for the test suites: [${filters.sorted}]")
-      case projects => Right(new TestSuiteDebugAdapter(projects, filters, state, ioScheduler))
+      case Seq(project) if project.platform.isInstanceOf[Platform.Jvm] =>
+        val Platform.Jvm(config, _, _, _, _, _) = project.platform
+        val classPathEntries = getClassPathEntries(state.build, project)
+        val javaRuntime = JavaRuntime(config.javaHome.underlying)
+        val evaluationClassLoader = getEvaluationClassLoader(project, state, ioScheduler)
+        Right(
+          new TestSuiteDebugAdapter(
+            projects,
+            filters,
+            classPathEntries,
+            javaRuntime,
+            evaluationClassLoader,
+            state,
+            ioScheduler
+          )
+        )
+      case _ =>
+        Right(
+          new TestSuiteDebugAdapter(projects, filters, Seq.empty, None, None, state, ioScheduler)
+        )
+    }
+  }
+
+  def forAttachRemote(
+      state: State,
+      ioScheduler: Scheduler,
+      projects: Seq[Project]
+  ): DebuggeeRunner = {
+    projects match {
+      case Seq(project) if project.platform.isInstanceOf[Platform.Jvm] =>
+        val Platform.Jvm(config, _, _, _, _, _) = project.platform
+        val classPathEntries = getClassPathEntries(state.build, project)
+        val javaRuntime = JavaRuntime(config.javaHome.underlying)
+        val evaluationClassLoader = getEvaluationClassLoader(project, state, ioScheduler)
+        new AttachRemoteDebugAdapter(
+          classPathEntries,
+          javaRuntime,
+          evaluationClassLoader,
+          state,
+          ioScheduler
+        )
+      case _ => new AttachRemoteDebugAdapter(Seq.empty, None, None, state, ioScheduler)
     }
   }
 
-  def forAttachRemote(state: State, ioScheduler: Scheduler): DebuggeeRunner =
-    new AttachRemoteDebugAdapter(state, ioScheduler)
+  private def getClassPathEntries(build: Build, project: Project): Seq[ClassPathEntry] = {
+    val allProjects = getAllDepsRecursively(build, project)
+    getLibraries(allProjects) ++ getClassDirectories(allProjects)
+  }
+
+  private def getEvaluationClassLoader(
+      project: Project,
+      state: State,
+      ioScheduler: Scheduler
+  ): Option[ClassLoader] = {
+    project.scalaInstance
+      .flatMap(getEvaluationClassLoader(_, state.logger, ioScheduler))
+  }
+
+  private def getEvaluationClassLoader(
+      scalaInstance: ScalaInstance,
+      logger: Logger,
+      ioScheduler: Scheduler
+  ): Option[ClassLoader] = {
+    import ch.epfl.scala.debugadapter.BuildInfo._
+    import coursier._
+    val scalaVersion = scalaInstance.version
+    val module = s"${expressionCompilerName}_$scalaVersion"
+    val expressionCompilerDep = Dependency(
```

Comment:
Could we cache it similar to how we do it with semanticdb plugins? Take a look at `SemanticDBCache` for details. This way we will not attempt to reresolve if there are issues.

Reply:
Done! Thanks for the pointer.